### ISSUE TITLE
Contact Form Gone after latest update!

### ISSUE THREAD

**Issue by axel-rank** [getgrav/grav-plugin-form#217]
After latest Form update to 2.11.0 my Contact Form is not showing anymore!
(It took me a while to set it up and got it working...!)
What did you change???
My setup is like described in the docs...
I have a modular form.
form is defined in modular.md

**Comment by axel-rank** [comment]
Just reinstalled 2.10 and everythings ok again.
Installed 2.11 --> contact form gone.
Copied **form.php** from 2.10 to 2.11 --> contactform showing again!
So changes in form.php causes my form to dissapear.

**Comment by axel-rank** [comment]
Ok, the form definition has to be in the page header of the contactpage not in modular.md.
Maybe you can update the docs or give a message next time if you do such kind of "optimizations".
Thank you.

**Comment by rhukster** [comment]
Can you provide me with a zip of the pages that can alllow me to test this? I have tested a modular form and it works fine but I don’t know how others set these things up.

**Comment by Balindil** [comment]
Same issue here.

**Comment by rhukster** [comment]
Same response to you @Balindil .  Need an example I can test.

**Comment by Balindil** [comment]
You can see it here: [https://demo.getgrav.org/agency-skeleton/](https://demo.getgrav.org/agency-skeleton/)
At the bottom of the page should be the contact form.

**Comment by rhukster** [comment]
Agency demo was out of date using the old `simple_form` plugin.  I updated it to latest using latest `form` plugin and the form shows.

The form even works, although all the labels and the message itself is rendered in grey text on grey background so impossible to read.  That needs fixing, but the form itself is fine.

**Comment by rhukster** [comment]
@axel-rank please provide me a sample that is NOT working..  Cheers.

**Comment by axel-rank** [comment]
I'm using theme inheritance based on agency template.
Form shows only if I put form definition in _contact/form.md. If it's in modular.md form doesn't show, which worked until Form 2.11.0.
Maybe it's related to form.html.twig.
form.md:
`---
title: Kontakt
form:
    name: contact
    action: /
    fields:
        -
            name: name
            label: Name
            classes: form-control
            placeholder: 'Name eingeben'
            type: text
            position: left
            validate:
                required: true
        -
            name: email
            label: Email
            classes: form-control
            placeholder: 'Email Adresse eingeben'
            type: email
            position: left
            validate:
                required: true
                pattern: "[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$"
        -
            name: message
            label: Nachricht
            placeholder: 'Nachricht schreiben'
            type: textarea
            classes: form-control
            position: right
            validate:
                required: true
    buttons:
        -
            type: submit
            classes: 'btn btn-primary btn-lg'
            value: Senden
    process:
        -
            email:
                from: '{{ form.value.email }}'
                subject: '[Homepage Kontakt von] {{ form.value.name|e }}'
                body: '{% include ''forms/data.html.twig'' %}'
        -
            reset: true
        -
            message: 'Vielen Dank für die Nachricht! Ich melde mich so schnell wie möglich.'

cache_enable: false
---`
modular.md:
`---
title: Musik und Gitarrenunterricht im Fichtelgebirge, Bayreuth und Pegnitz
menu: Home
onpage_menu: true
content:
    items: '@self.modular'
    order:
        by: default
        dir: asc
        custom:
            - _header
            - _news
            - _gitarre
            - _musik
            - _pics
            - _uebermich
            - _links
            - _contact
            - _impressum

---
`
form.html.twig:
`<section id="contact">
  <div class="container">
    <div class="row">
      <div class="col-lg-12 text-center">
        {{ content }}

{% set scope = scope ?: 'data.' %}
{% set multipart = '' %}
{% set method = form.method|upper|default('POST') %}

{% for field in form.fields %}
    {% if (method == 'POST' and field.type == 'file') %}
        {% set multipart = ' enctype="multipart/form-data"' %}
    {% endif %}
{% endfor %}

{% set action = form.action ? base_url ~ form.action : base_url ~ page.route %}

{% if (action == base_url_relative) %}
    {% set action = base_url_relative ~ '/' ~ page.slug %}
{% endif %}

<form name="{{ form.name }}"
      action="{{ action }}"
      method="{{ method }}"{{ multipart }}
      {% if form.id %}id="{{ form.id }}"{% endif %}
      {% block form_classes %}
      {% if form.classes %}class="{{ form.classes }}"{% endif %}
      {% endblock %}
>

  {% block inner_markup_fields_start %}{% endblock %}

  <div class="col-md-6">
    {% for field in form.fields %}
      {% if field.position == 'left' %}
        {% set value = form.value(field.name) %}
        <div class="form-group">
          {% include "forms/fields/#{field.type}/#{field.type}.html.twig" ignore missing %}
        </div>
      {% endif %}  
    {% endfor %}
  </div>

<div class="col-md-6">
    {% for field in form.fields %}
      {% if field.position == 'right' %}
        {% set value = form.value(field.name) %}
        <div class="form-group">
          {% include "forms/fields/#{field.type}/#{field.type}.html.twig" ignore missing %}
        </div>
      {% endif %}  
    {% endfor %}
  </div>



  {% include "forms/fields/formname/formname.html.twig" %}

  {% block inner_markup_fields_end %}{% endblock %}



  {% block inner_markup_buttons_start %}
  <div class="buttons">
  {% endblock %}

  {% for button in form.buttons %}
      <button
            {% block button_classes %}
            class="{{ button.classes|default('button') }}"
            {% endblock %}
            type="{{ button.type|default('submit') }}"
        >
            {{ button.value|t|default('Submit') }}
        </button>
  {% endfor %}

  {% block inner_markup_buttons_end %}
  </div>
  {% endblock %}

  {{ nonce_field('form', 'form-nonce')|raw }}
</form>
      
        
      </div>
    </div></div>
    
</section>
`

**Comment by axel-rank** [comment]
I'm using theme inheritance based on agency theme.
Simple Form plugin isn't installed.
Form only shows if form definition is in page header of form.md
If form definition is in modular.md form doesn't show anymore in Form 2.11.0
Maybe it's related to form.html.twig

My form.md in 01.home/_contact:

```
---
title: Kontakt
form:
    name: contact
    action: /
    fields:
        -
            name: name
            label: Name
            classes: form-control
            placeholder: 'Name eingeben'
            type: text
            position: left
            validate:
                required: true
        -
            name: email
            label: Email
            classes: form-control
            placeholder: 'Email Adresse eingeben'
            type: email
            position: left
            validate:
                required: true
                pattern: "[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$"
        -
            name: message
            label: Nachricht
            placeholder: 'Nachricht schreiben'
            type: textarea
            classes: form-control
            position: right
            validate:
                required: true
    buttons:
        -
            type: submit
            classes: 'btn btn-primary btn-lg'
            value: Senden
    process:
        -
            email:
                from: '{{ form.value.email }}'
                subject: '[Homepage Kontakt von] {{ form.value.name|e }}'
                body: '{% include ''forms/data.html.twig'' %}'
        -
            reset: true
        -
            message: 'Vielen Dank für die Nachricht! Ich melde mich so schnell wie möglich.'

cache_enable: false
---

```
modular.md:

```
---
title: Musik und Gitarrenunterricht im Fichtelgebirge, Bayreuth und Pegnitz
menu: Home
onpage_menu: true
content:
    items: '@self.modular'
    order:
        by: default
        dir: asc
        custom:
            - _header
            - _news
            - _gitarre
            - _musik
            - _pics
            - _uebermich
            - _links
            - _contact
            - _impressum

---
```
form.html.twig:
```
<section id="contact">
  <div class="container">
    <div class="row">
      <div class="col-lg-12 text-center">
        {{ content }}

{% set scope = scope ?: 'data.' %}
{% set multipart = '' %}
{% set method = form.method|upper|default('POST') %}

{% for field in form.fields %}
    {% if (method == 'POST' and field.type == 'file') %}
        {% set multipart = ' enctype="multipart/form-data"' %}
    {% endif %}
{% endfor %}

{% set action = form.action ? base_url ~ form.action : base_url ~ page.route %}

{% if (action == base_url_relative) %}
    {% set action = base_url_relative ~ '/' ~ page.slug %}
{% endif %}

<form name="{{ form.name }}"
      action="{{ action }}"
      method="{{ method }}"{{ multipart }}
      {% if form.id %}id="{{ form.id }}"{% endif %}
      {% block form_classes %}
      {% if form.classes %}class="{{ form.classes }}"{% endif %}
      {% endblock %}
>

  {% block inner_markup_fields_start %}{% endblock %}

  <div class="col-md-6">
    {% for field in form.fields %}
      {% if field.position == 'left' %}
        {% set value = form.value(field.name) %}
        <div class="form-group">
          {% include "forms/fields/#{field.type}/#{field.type}.html.twig" ignore missing %}
        </div>
      {% endif %}  
    {% endfor %}
  </div>

<div class="col-md-6">
    {% for field in form.fields %}
      {% if field.position == 'right' %}
        {% set value = form.value(field.name) %}
        <div class="form-group">
          {% include "forms/fields/#{field.type}/#{field.type}.html.twig" ignore missing %}
        </div>
      {% endif %}  
    {% endfor %}
  </div>



  {% include "forms/fields/formname/formname.html.twig" %}

  {% block inner_markup_fields_end %}{% endblock %}



  {% block inner_markup_buttons_start %}
  <div class="buttons">
  {% endblock %}

  {% for button in form.buttons %}
      <button
            {% block button_classes %}
            class="{{ button.classes|default('button') }}"
            {% endblock %}
            type="{{ button.type|default('submit') }}"
        >
            {{ button.value|t|default('Submit') }}
        </button>
  {% endfor %}

  {% block inner_markup_buttons_end %}
  </div>
  {% endblock %}

  {{ nonce_field('form', 'form-nonce')|raw }}
</form>
      
        
      </div>
    </div></div>
    
</section>

```

**Comment by rhukster** [comment]
I'm not able to reproduce this error with Agency theme. Being an inherited theme should make no difference.  I moved the form definition from the `modular.md` to the `form.md` file with no other edits and it worked fine.  I did actually modify the `templates/modular.html.twig` slightly from the latest in develop and updated it so that it referenced the `partials/form-messages.html.twig` template rather than outputing the messages itself.  Other than that it's unchanged:

https://github.com/getgrav/grav-theme-agency/commit/90fda59fe2a22e706bee09b21ab7099f6ae7245b

If anything, I think the form override might be out of date in Agency, so i just updated it to:

```
<section id="contact">
  <div class="container">
    <div class="row">
      <div class="col-lg-12 text-center">
        {{ content }}
      </div>
    </div>
    <div class="row">

    {% include "forms/form.html.twig" %}

    </div>
  </div>
</section>
```

And while styling wasn't as great, it worked fine.
